src/domain.ts

```diff
@@ -0,0 +1,39 @@
+import { create } from 'node:domain'
+import { CollectionTag, Schema, YAMLMap } from 'yaml'
+
+const tag = '!domain'
+
+class YAMLDomain extends YAMLMap {
+  tag = tag
+
+  toJSON() {
+    return create()
+  }
+
+  // no way to faithfully return a proper domain, and capturing
+  // pretty much any of its properties is a nightmare, so just
+  // leave it empty.
+  static from(schema?: Schema) {
+    return new this(schema)
+  }
+}
+
+/**
+ * `!domain` Domain
+ *
+ * [Node.js Domain](https://nodejs.org/api/domain.html) values.
+ * No properties are preserved, because their huge object graphs
+ * can be very costly to serialize, and are not particularly useful.
```

Comment:
In that case I'd prefer to leave domains out, at least for now.

Reply:
Just closing the loop on this, it turns out that because the new version of tap (which will use this library) will actually break pretty badly if `require('domain')` is ever called, because it has a few side effects that don't play nice with node's newer error handling interfaces. So the chances of anyone needing this in their test report output is basically zero, and removing it is even more reasonable. If they have domains in play, other stuff is already blowing up in their face 🤣

Reply:
I was just about to add a 👍 to that, but then I realised that would've been a bit weird. So a 👍 to not needing to add domains here, but sympathies and commiserations to any thus affected.